### ISSUE TITLE
Group subsequent records into a single row

### ISSUE THREAD

**Issue by fjarrett** [xwp/stream#516]
Originally raised by @topher1kenobe during our Austin meetup, and again formally by @Japh in #474, was the concern that the current implementation of the Stream list table screen does nothing to mitigate the flooding of records of the same type.

After discussing this with @lukecarbis, we think the solution could be achieved with a smarter UI that catches subsequent records that have the same Author/Connector/Context/Action and grouping them. The freshest record would always be visible, and additional records could be expanded if desired.

**CURRENT EXPERIENCE**

![screen shot 2014-05-07 at 4 40 10 pm](https://cloud.githubusercontent.com/assets/522158/2909793/a22fb2b0-d637-11e3-8356-2c90bedc9484.png)

**PROPOSED SOLUTION**

![screen shot 2014-05-07 at 5 32 05 pm](https://cloud.githubusercontent.com/assets/522158/2909794/a648947a-d637-11e3-9c0b-cf85ee675b06.png)

**IMPROVED SCREEN OPTIONS**

![screen_shot_2014-05-07_at_5_43_30_pm](https://cloud.githubusercontent.com/assets/522158/2909865/2329b6f8-d639-11e3-9e5b-386cacca5ec5.png)

**Comment by fjarrett** [comment]
FYI I'm going to revisit this issue soon.

Now that we have the power of Elasticsearch on our side we can accomplish much more of what this issue needs in regards to wildcard queries, aggregation and pagination.

**Comment by lukecarbis** [comment]
Now that we _don't_ have the power of Elasticsearch anymore, I'd like to revisit this issue. 

There are two ways to group records: During the DB Query (or with PHP post query), or with Javascript.

The sort of MySQL query that could achieve this would be extremely expensive, as would any post processing.

Javascript has problems, too. You run into pagination issues. Suppose records 1-6 should be grouped, and records 7-15 should be grouped. You are only showing 10 records on the page, so you get 1-6 grouped and 7-10 grouped, then when you go to the next page you get 11-15 grouped separately.

Plus, in this example, you only have two entries on the page, where you should have 10.

The only way I can think to solve this is with a complicated AJAX setup, which continues to load until you've got everything, but this has problems too - if you need to group 500 records together, the page would take forever to finish loading.

In other words, I think it's worth doing, I just don't know how.

**Comment by shadyvb** [comment]
I think the answer actually might be in *pre-processing* rather than *post-processing*, ie: caching last record, and comparing with the new one to be inserted, probably using the first post's ID as a shared key with all subsequent entries, which you can later group with.

Risks might involve racing conditions, but we experiment about doing it in SQL, not sure how yet. Either duplicating writing queries, one for the log and another for the grouping, or a manually initiated / cron initiated cleanup process that does this.

**Comment by shadyvb** [comment]
Experiment nested write queries, checking for the last recorded log ( via having no WHERE but a LIMIT 1 ), and using IF to use the found ID in post_parent.

**Comment by lukecarbis** [comment]
Closing. Planning to address this in Stream 4.
